logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInTheConversationUseCase.kt

```diff
@@ -0,0 +1,32 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.publicuser.model.OtherUser
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.functional.map
+
+class GetAllContactsNotInTheConversationUseCase(
+    private val conversationRepository: ConversationRepository,
+    private val userRepository: UserRepository
+) {
+    suspend operator fun invoke(conversationId: QualifiedID) =
+        conversationRepository.getConversationMembers(conversationId).flatMap { conversationMembers ->
+            userRepository.getAllContacts()
+                .map { allContacts -> removeConversationMembersFromAllContacts(allContacts, conversationMembers) }
+        }.fold({ Result.Failure(it) }, { Result.Success(it) })
+
+    private fun removeConversationMembersFromAllContacts(allContacts: List<OtherUser>, conversationMembers: List<UserId>) =
+        allContacts.filter { conversationMember -> !conversationMembers.contains(conversationMember.id) }
+}
```

Comment:
i wonder if doing this on the sql level will be faster than iterating over 2 whole lists to filter.

Reply:
Valid Point ill look into

Reply:
will make a change and submit the change soon